tests/bsim/bluetooth/mesh/tests_scripts/persistence/provisioning.sh

```diff
@@ -4,14 +4,35 @@
 
 source $(dirname "${BASH_SOURCE[0]}")/../../_mesh_test.sh
 
-# Note:
+# Test checks that mesh provisioning data is saved and loaded correctly.
+# It also checks if some key exists in the PSA ITS, it is destroyed and
+# the correct key is used instead.
+#
 # Tests must be added in pairs and in sequence.
 # First test: saves data; second test: verifies it.
+#
+# Test procedure:
+# 0. Fake key is created in the PSA ITS (only if the fake key is used).
+# 1. Device Under Test (DUT) initializes the Mesh stack,
+#    and starts provisioning and configuration itself.
+# 2. Test scenario emulates hardware  reset by running the second test
+#    with stored data from the first test.
+# 3. DUT checks that the fake key was substituted by the real key (only if the fake key is used).
```

Comment:
What exactly you mean by 'fake key' ? Do you mean to say 'garbage key' or 'stale key'? If yes, then it is worth to use this term instead of using fake key term.